File: 2574-left-and-right-sum-differences/2574-left-and-right-sum-differences.cpp
class Solution {
public:
    vector<int> leftRightDifference(vector<int>& nums) {
        stack<int>st;
        queue<int>pq;
        for(auto i:nums){
            st.push(i);
            pq.push(i);
        }
        vector<int>ls;
        vector<int>rs;
        int prev=0;
        int aft=0;
        for(int i=0;i<nums.size();i++){
            ls.push_back(prev);
            prev+=pq.front();
            pq.pop();
            rs.push_back(aft);
            aft+=st.top();
            st.pop();
        }
        reverse(rs.begin(),rs.end());
        vector<int>ans;
        for(int i=0;i<nums.size();i++){
            ans.push_back(abs(ls[i]-rs[i]));
        }
        return ans;
    }
};

File: 2574-left-and-right-sum-differences/2574-left-and-right-sum-differences.py
class Solution:
    def leftRightDifference(self, nums: List[int]) -> List[int]:
        p=[0]*len(nums)
        s=[0]*len(nums)
        for i in range(1,len(nums)):
            p[i]=p[i-1]+nums[i-1]
        for i in range(len(nums)-2,-1,-1):
            s[i]=s[i+1]+nums[i+1]
        ans=[0]*len(nums)
        for i in range(len(nums)):
            ans[i]=abs(p[i]-s[i])
        return ans
            